Src/StackifyLib/Config.cs

```diff
@@ -197,5 +216,137 @@ internal static string Get(string key, string defaultValue = null)
+                    {
+                        using (var fs = new FileStream(iisJsonPath, FileMode.Open, FileAccess.Read, FileShare.ReadWrite))
+                        {
+                            using (var sr = new StreamReader(fs))
+                            {
+                                json = sr.ReadToEnd();
+                            }
+                        }
+
+                        var obj = JObject.Parse(json, Settings);
+                        Config.SetStackifyObj(obj);
+                    }
+                }
+#endif
+            }
+            catch (Exception ex)
+            {
+                StackifyAPILogger.Log("#Config #ReadStackifyJSONConfig failed", ex);
+            }
+            IsStackifyJsonLoaded = true;
+        }
+
+#if JSONTEST
+        public static void ReadStackifyJSONConfig(string filePath)
+        {
+            try
+            {
+                string json;
+
+                using (var fs = new FileStream(filePath, FileMode.Open, FileAccess.Read, FileShare.ReadWrite))
+                {
+                    using (var sr = new StreamReader(fs))
+                    {
+                        json = sr.ReadToEnd();
+                    }
+                }
+
+                var obj = JObject.Parse(json, Settings);
+                Config.SetStackifyObj(obj);
+            }
+            catch (Exception ex)
+            {
+                StackifyAPILogger.Log("#Config #ReadStackifyJSONConfig failed", ex);
+            }
+            
+        }
+#endif
+
+        public static void SetStackifyObj(JObject obj)
+        {
+            AppName = TryGetValue(obj, "AppName") ?? AppName;
+            Environment = GetEnvironment(obj);
+            ApiKey = TryGetValue(obj, "ApiKey") ?? ApiKey;
+        }
+
+        private static string TryGetValue(JToken jToken, string key)
+        {
+            string r = null;
+
+            try
+            {
+                var val = jToken[key];
+                if (val == null)
+                {
+                    StackifyAPILogger.Log($"#Config #TryGetValue #Json failed - Property is null or empty - Property: {key}");
+                    return r;
+                }
+
+                if (val.Type != JTokenType.String || (val.Type == JTokenType.String && string.IsNullOrEmpty(val.ToString())))
+                {
+                    StackifyAPILogger.Log($"#Config #TryGetValue #Json failed - Property is not a string - Property: {key}");
+                    return r;
+                }
+
+                r = val.ToString();
+            }
+            catch (Exception ex)
+            {
+                StackifyAPILogger.Log("#Config #TryGetValue #Json failed", ex);
+            }
+
+            return r;
+        }
+        public static string GetEnvironment(JObject envName = null)
+        {
+            var ASPEnvironment = System.Environment.GetEnvironmentVariable("ASPNETCORE_ENVIRONMENT");
```

Comment:
These environment variables should be used to load additional overrides for the Stackify.json.  For example, if DOTNET_ENVIRONMENT is set to "Production" then both Stackify.json and Stackify.Production.json should be read and the values in Stackify.Production.json should replace those in Stackify.json (or add to to the values if the value was not in Stackify.json.  These are not intended to set the application environment to reporting to Retrace (at least not directly).  There are environment variables STACKIFY_APPNAME and STACKIFY_ENVIRONMENT that are used to override the AppName and Environment reported to Retrace supported in the wrappers, and these should also be supported in StackifyLib.

Reply:
added support on which environment variable used to load the additional overrides for Stackify.json